src/deterministic_vi/elbo.jl

```diff
@@ -406,20 +408,27 @@ function populate_fsm_vecs!{NumType <: Number}(
         calculate_hessian =
             elbo_vars.calculate_hessian && elbo_vars.calculate_derivs && active_source
         clear!(elbo_vars.fs0m_vec[s], calculate_hessian)
-        for k = 1:psf_K # PSF component
-            accum_star_pos!(
-                elbo_vars, s, star_mcs[k, s], x, wcs_jacobian, active_source)
+        for k = 1:ea.psf_K # PSF component
+            if (ea.num_allowed_sd == Inf ||
+                check_point_close_to_bvn(star_mcs[k, s], x, ea.num_allowed_sd))
```

Comment:
Isn't this going to lead to discontinuities? I think we want to continue to fix the subimage around each celestial body before the optimization.